pkg/logs/sender/sender.go

```diff
@@ -7,55 +7,64 @@ package sender
 
 import (
 	"context"
+	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/logs/client"
+	"github.com/DataDog/datadog-agent/pkg/logs/client/tcp"
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
 )
 
-// Sender is responsible for sending logs to different destinations.
-type Sender struct {
+// Sender sends logs to different destination and reports to auditor
+type Sender interface {
```

Comment:
it feels like there is an implementation per protocol so I would put the tcp logic under a `tcp` package and the `http` logic under a `http` package